valhalla/filesystem.h

```diff
@@ -491,4 +494,65 @@ inline std::chrono::time_point<std::chrono::system_clock> last_write_time(const
   return std::chrono::system_clock::from_time_t(FS_MTIME(s));
 }
 
+/**
+ * @brief Saves data to the path.
+ * @attention Will replace the contents in case if fpath already exists. Will create
+ * new directory if directory did not exist before hand.
+ * */
+template <typename T> inline bool save(const std::string& fpath, const T& data = {}) {
```

Comment:
You're assuming that data is a container with sequential memory layout(e.g, vector, array). Does it make sense to specify it explicitly?
```suggestion
template <typename T> inline bool save(const std::string& fpath, const vector<T>& data = {}) {
```

Reply:
needed it here https://github.com/valhalla/valhalla/blob/3146fc64fbb992a1754516f9bf8a7de4a018df0c/test/elevation_builder.cc#L407